packages/monitor-v2/src/llm-bot/common.ts

```diff
@@ -291,3 +338,17 @@ export interface OptimisticOracleClientFilter<I extends OptimisticOracleRequest,
    */
   filter(optimisticOracleRequests: I[]): Promise<O[]>;
 }
+
+/**
+ * Abstract class representing a strategy for processing Optimistic Oracle requests.
+ * @template I The type of the input OptimisticOracleRequest.
+ * @template R The type of the result, based on OptimisticOracleRequestDisputable.
+ */
+export interface LLMDisputerStrategy<I extends OptimisticOracleRequest, R extends OptimisticOracleRequestDisputable> {
```

Comment:
I really like how simple these classes are, while keeping the flexibility to pass in more complex types of requests.